### PR TITLE
feat(azure-iot-hub): finalize pnp apis

### PR DESCRIPTION
**Pull request by zolvarga**
# Checklist
- [ ] I have read the [contribution guidelines] (https://github.com/Azure/azure-iot-sdk-node/blob/master/.github/CONTRIBUTING.md).
- [ ] I added or modified the existing tests to cover the change (we do not allow our test coverage to go down).
- If this is a modification that impacts the behavior of a public API
  - [ ] I edited the corresponding document in the `devdoc` folder and added or modified requirements.

# Reference/Link to the issue solved with this PR (if any)

# Description of the problem

# Description of the solution

### CODE REVIEW

File: digitaltwins/samples/service/javascript/get_digital_twin.js [hunk 1]
```diff
@@ -19,11 +19,11 @@ async function main() {
   const credentials = new IoTHubTokenCredentials(process.env.IOTHUB_CONNECTION_STRING);
   const digitalTwinServiceClient = new DigitalTwinServiceClient(credentials);
 
-  // Get digital twin
+  // Get digital twin and retrieve the modelId from it
   const digitalTwin = await digitalTwinServiceClient.getDigitalTwin(deviceId);
 
-  // Print digital twin
   console.log(inspect(digitalTwin));
+  console.log("Model Id: " + inspect(digitalTwin.$metadata.$model))
```
Comment (anthonyvercolano) [inline digitaltwins/samples/service/javascript/get_digital_twin.js]: >) [](start = 66, length = 1)

OMG  Look at this.  You let a semi-colon get away.  It's probably going to hide under your desk and reproduce.  It's very expensive at that point.

File: digitaltwins/service/src/cl/digital_twin_service_client.ts [hunk 2]
```diff
@@ -191,114 +138,28 @@ export class DigitalTwinServiceClient {
...
-    /*Codes_SRS_NODE_DIGITAL_TWIN_SERVICE_CLIENT_12_009: [The `getModel` method shall call the callback with an error parameter if a callback is passed..]*/
-    /*Codes_SRS_NODE_DIGITAL_TWIN_SERVICE_CLIENT_12_010: [The `getModel` method shall return error if the method of the protocol layer failed.]*/
-    /*Codes_SRS_NODE_DIGITAL_TWIN_SERVICE_CLIENT_12_022: [The `getModel` method shall return a promise if there is no callback passed.]*/
-    return tripleValueCallbackToPromise<Model, msRest.HttpOperationResponse, DigitalTwinGetDigitalTwinModelResponse>((_callback) => {
-      this._pl.digitalTwin.getDigitalTwinModel(modelId, (err, result, request, response) => {
-        _callback(err as Error, result, response);
-      });
-    }, (result, response) => createResultWithHttpOperationResponse<Model, DigitalTwinGetDigitalTwinModelResponse>(result, response), callback as TripleValueCallback<Model, msRest.HttpOperationResponse>);
+  invokeComponentCommand(digitalTwinId: string, componentName: string, commandName: string, argument: string): Promise<DigitalTwinInvokeComponentCommandResponse>{
```
Comment (anthonyvercolano) [inline digitaltwins/service/src/cl/digital_twin_service_client.ts:150]: >invokeComponentCommand [](start = 2, length = 22)

They didn't balk at two api for invoking commands?

Comment (zolvarga) [reply digitaltwins/service/src/cl/digital_twin_service_client.ts:150]: We have invokeComponentCommand and invokeCommand. Is that answers your question?

---
In reply to: [452337338](https://github.com/Azure/azure-iot-sdk-node/pull/844#discussion_r452337338) [](ancestors = 452337338)

File: digitaltwins/service/test/_digital_twin_service_client_test.js [hunk 3]
```diff
@@ -236,367 +236,38 @@ describe('DigitalTwinServiceClient', function () {
     assert.deepEqual(returnedPromise._response, testUserPatch);
   });
 
-  /* Tests_SRS_NODE_DIGITAL_TWIN_SERVICE_CLIENT_12_005: [The `getComponents` method shall call the `getComponents` method of the protocol layer with the given arguments.]*/
-  /* Tests_SRS_NODE_DIGITAL_TWIN_SERVICE_CLIENT_12_006: [The `getComponents` method shall call the callback with an error parameter if a callback is passed..]*/
-  it('getComponents calls getComponents on the PL client', function (testCallback) {
+  it('invokeComponentCommand called with the right parameters and shall return a promise', async () => {
```
Comment (anthonyvercolano) [inline digitaltwins/service/test/_digital_twin_service_client_test.js:239]: >it( [](start = 2, length = 3)

No tests for invalid number of parameters?

Comment (zolvarga) [reply digitaltwins/service/test/_digital_twin_service_client_test.js:239]: Implicitly there is with the CalledWith

---
In reply to: [452338541](https://github.com/Azure/azure-iot-sdk-node/pull/844#discussion_r452338541) [](ancestors = 452338541)

**Pull request merged**